tar.go

```diff
@@ -29,11 +30,13 @@ func NewTarStreamer(r io.Reader, keywords []string) Streamer {
 		tarReader:  tar.NewReader(pR),
 		keywords:   keywords,
 	}
-	go ts.readHeaders() // I don't like this
+
+	go ts.readHeaders()
```

Comment:
Can someone explain to me why we implemented this interface as a reader? I don't understand that. Why not have a `Walk`-like function that does everything internally. Then we can just read from the data and parse it as we go -- creating the Hierarchy on the fly? Is it because of the ordering of entries again (does this actually matter for `mtree` because it's just a generic tagging system)?